crypto/dkg_feldmanvss.go

```diff
@@ -256,17 +267,23 @@ func (s *feldmanVSSstate) generateShares(seed []byte) error {
 	s.a = make([]scalar, s.threshold+1)
 	s.vA = make([]pointG2, s.threshold+1)
 	s.y = make([]pointG2, s.size)
-	randZrStar(&s.a[0]) // non zero a[0]
+	// non-zero a[0] - group private key is not zero
+	randZrStar(&s.a[0])
 	generatorScalarMultG2(&s.vA[0], &s.a[0])
-	for i := 1; i < s.threshold+1; i++ {
-		C.bn_new_wrapper((*C.bn_st)(&s.a[i]))
-		randZr(&s.a[i])
-		generatorScalarMultG2(&s.vA[i], &s.a[i])
+	if s.threshold > 0 {
+		for i := 1; i < s.threshold; i++ {
+			C.bn_new_wrapper((*C.bn_st)(&s.a[i]))
+			randZr(&s.a[i])
```

Comment:
Why do we allow maybe-zero values for `a[1:threshold]`, but not `a[0]` and `a[threshold]`? I'm also wondering why this differs from the [similar code block in threshold signing](https://github.com/onflow/flow-go/pull/3421/files#diff-09d2f77aa5f9d78dc6b7d19a2c640a9fb7710671880625092cc4b1cfece6bd67R575-R580), where we use non-zero values for all entries in `a`

Reply:
Good question. This is related to Shamir secret sharing, where we need to generate a polynomial `P` of degree `t` : P(x) = a_t.x^t + ... + a_1.x + a_0 
To make sure the polynomial is of degree `t` (and avoid possible security issues related to the threshold), we need to enforce `a_t` is non-zero (otherwise `P` is of degree at most `t-1` ).

Moreover, in Shamir secret sharing, the secret (here private key) is `a_0`. As part of this PR where we don't want to allow identity private keys and identity public keys, we want to enforce `a_0` is non-identity (non-zero). The rest of the values `a_i` do not matter.

Reply:
The other block in threshold signing is buggy https://github.com/onflow/flow-go/pull/3421#discussion_r1006000477